ValidationLibrary.Rules.Tests/HasReadmeRuleTests.cs

```diff
@@ -10,45 +10,17 @@
 
 namespace ValidationLibrary.Tests.Rules
 {
-    public class HasReadmeRuleTests
+    public class HasReadmeRuleTests : BaseRuleTests<HasReadmeRule>
     {
-        /// <summary>
-        /// By default, master is checked for README.md if there is no branch
-        /// </summary>
-        private const string MasterBranch = "master";
-
-        private HasReadmeRule _rule;
-
-        private User _owner;
-
-        private IGitHubClient _mockClient;
-        private IRepositoriesClient _mockRepositoryClient;
-        private IRepositoryContentsClient _mockRepositoryContentClient;
-
-        [SetUp]
-        public async Task Setup()
+        protected override void OnSetup()
         {
-            _owner = new User(null, null, null, 0, null, DateTime.UtcNow, DateTime.UtcNow, 0, null, 0, 0, false, null, 0, 0, null, "protacon", "protacon", null, 0, null, 0, 0, 0, null, new RepositoryPermissions(), false, null, null);
-
-            _mockClient = Substitute.For<IGitHubClient>();
-            _mockRepositoryClient = Substitute.For<IRepositoriesClient>();
-            _mockClient.Repository.Returns(_mockRepositoryClient);
-            _mockRepositoryContentClient = Substitute.For<IRepositoryContentsClient>();
-            _mockRepositoryClient.Content.Returns(_mockRepositoryContentClient);
-
             _rule = new HasReadmeRule(
                 Substitute.For<ILogger<HasReadmeRule>>(),
                 new GitUtils(Substitute.For<ILogger<GitUtils>>()));
 
             var mockReleaseClient = Substitute.For<IReleasesClient>();
-            _mockRepositoryClient.Release.Returns(mockReleaseClient);
-            await _rule.Init(_mockClient);
-        }
-
-        [Test]
-        public void RuleName_IsDefined()
-        {
-            Assert.NotNull(_rule.RuleName);
+            MockRepositoryClient.Release.Returns(mockReleaseClient);
+            _rule.Init(MockClient).Wait();
```

Comment:
Maybe the _rule.Init() should go in the base class so it acts for all the tests and is more generic, even if the other tests might not require the init() to execute.

Reply:
Good idea!